upup/pkg/fi/dryrun_target.go

```diff
@@ -186,14 +190,22 @@ func (t *DryRunTarget) PrintReport(taskMap map[string]Task, out io.Writer) error
 							switch fieldValE.Interface().(type) {
 							//case SimpleUnit:
 							//	ignored = true
-							default:
+							case Resource, ResourceHolder:
+								resA, okA := tryResourceAsString(fieldValA)
+								resE, okE := tryResourceAsString(fieldValE)
+								if okA && okE {
+									description = utils.FormatDiff(resA, resE)
+								}
+							}
+
+							if !ignored && description == "" {
 								description = fmt.Sprintf(" %v -> %v", ValueAsString(fieldValA), ValueAsString(fieldValE))
 							}
 						}
 						if ignored {
 							continue
 						}
-						changeList = append(changeList, fmt.Sprintf("%-20s\t%s", valC.Type().Field(i).Name, description))
+						changeList = append(changeList, change{FieldName: valC.Type().Field(i).Name, Description: description})
```

Comment:
should we be checking `valC.Type().Field(i).Name` to see if the value exists?  Seems like a lot of possibilities for a panic.